src/kaleidoscope/hid.cpp

```diff
@@ -51,13 +51,86 @@
 namespace kaleidoscope {
 namespace hid {
 
+// Anonymous namespace for internal variables and helper functions
+namespace {
+
+// The functions in this namespace are primarily to solve the problem of rollover from a
+// key with a modifier flag (e.g. `LSHIFT(Key_T)`) to one without (e.g. `Key_H`), which
+// used to result in the mod flag being applied to keys other than the one with the
+// flag. By using `mod_flags_allowed`, we can prohibit any modifier flags that aren't in
+// the key most recently pressed, fixing the rollover problem, and getting the intended
+// `The` instead of `THe`. This was particularly noticeable to people who were using a
+// separate layer for `shift`, and a OneShot layer key to activate it, causing what looked
+// like a OneShot bug.
+
+// A bitfield of the current modifier flags from keys processed so far in the current
+// keyswitch scan:
+static byte current_mod_flags{0};
```

Comment:
I'm trying to think about naming that makes it obvious what this variable means.

Maybe something like modifiers_to_report? That still feels wrong and may be evidence that I don't understand the code yet.

Reply:
How about `scanned_mod_flags`, or something of the sort? To me, `modifiers_to_report` isn't great. The name should indicate that it's only modifier _flags_ that it stores, not ones from dedicated modifier keycodes. Also, on first reading "to report", my thought was that the "to" was indicating a destination, rather than a time delay (I read "report" as a noun, not a verb).